setup.py

```diff
@@ -19,7 +19,7 @@
     packages=setuptools.find_packages(exclude=['benchmarks', 'tests', 'tests.*']),
     install_requires=(
         'dnspython >= 1.15.0',
-        'greenlet >= 0.3',
+        'greenlet >= 1.0',
```

Comment:
Oh, 1.0.0 was more recent than I'd realized -- Jan 2021. Still, it's a couple years old; seems like a reasonable up-rev. *I'll* still want to watch out for it, but I'm stuck running old software (like py2) :stuck_out_tongue:

And yeah, once we've got that, `preserves_excinfo` would always be `True` in the old code, so we can rip all the workarounds out.